### DISCUSSION TITLE
Meta tags not in _document.js not working

### DISCUSSION THREAD

**Discussion by akanz** [vercel/next.js#35568] [Help] [answered]
Hello guys, I have been having issues with meta tags in my app. only the head tag in _document that's picked up. If i inspect the page I see the meta tags, but the meta tags doesn't show up when I use a meta preview (opengraph.xyz) or twitter card validator. 

I also noticed other head tags in doesn't appear in the source code but is present when I inspect the page. I've tried several solutions but none seem to work. Snippets of my code is present below: 

The first image is the _app.js

<img width="1440" alt="Screenshot 2022-03-24 at 12 18 31" src="https://user-images.githubusercontent.com/33222698/159906142-592c769a-a2bc-473a-907b-0b362deec866.png">

This is for SSR(Dynamic meta tag)

<img width="1310" alt="Screenshot 2022-03-24 at 12 20 08" src="https://user-images.githubusercontent.com/33222698/159906024-e7a807bb-0525-4c40-aec8-613dae2de91f.png">

**Comment by MamounDebbarh** [comment]
Hey! I am having the same issue. Let us know when you find out what's going on I think there are a lot of people struggling with this. From what I read it might deal with nextjs meta tags not being detected in children components for some reason. But couldn't find a fix.

**Reply by akanz** [reply]
Like what @dan-made said. SSR, Meta tags shouldn't be constrained to client side logic. appLoading was affecting it . I had to refactor the code for the meta tags and now it works fine.

**Comment by dan-made** [comment, marked as answer]
In your _app.js setup the only thing the only HTML that will get SSR'd is the splash screen. Your meta tags won't show up until setAppLoading is true. You'll need to refactor this to allow the page, or at least meta elements, to be available immediately.

**Reply by akanz** [reply]
Yes, you're absolutely correct. the client side logic affected the meta tags

**Reply by alanvera48** [reply]
Hi I have the same issue but in one my pages inside pages. I have to show a product, and i am in the [productId].js and i get the product information from the server. How can i pass the meta tags with the product title and image from the server?

**Reply by akanz** [reply]
No client side logic should influence the data you’re passing as the meta tag props. So you have to extract what you want as your meta data before any client side logic.
